src/pages/Blog/index.tsx

```diff
@@ -0,0 +1,399 @@
+import React, { useEffect, useState } from "react";
+import useStyles from "./styles";
+import { useTranslation } from "react-i18next";
+import {
+  Avatar,
+  Button,
+  Card,
+  CardActions,
+  CardContent,
+  CardMedia,
+  Container,
+  createStyles,
+  Grid,
+  Paper,
+  Tab,
+  Tabs,
+  Theme,
+  Typography,
+  withStyles,
+} from "@material-ui/core";
+import Footer from "../../components/Footer";
+import ReactMarkdown from "react-markdown";
+import { BLOG_KEYWORDS, VIEW_PORT } from "../../constants";
+import Sponsor from "../../components/Sponsor";
+import CustomTag from "../../components/CustomTag";
+import Pagination from "@material-ui/lab/Pagination";
+import blogs from "../../posts.json";
+
+interface StyledTabProps {
+  label: string;
+  value: string;
+}
+
+interface TabProps {
+  id: string;
+  title: string;
+  blog: string;
+  description: string;
+  image: string;
+  tags: string;
+  author: string;
+  length: number;
+}
+
+const Blog: React.FC = () => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const [jsonMdData, setJsonMdData] = useState<any>();
+  const [value, setValue] = React.useState("all");
+  const params = new URLSearchParams(window.location.search);
+  const queryAuthorName = params.get('author');
+  const queryTitleName = params.get('title');
+  const queryTagName = params.get('tags');
+  const mobileBreakpoint = VIEW_PORT.MOBILE_BREAKPOINT;
+  const itemsPerPage = 6;
+  const [page, setPage] = React.useState<number>(1);
+
+  const StyledTab = withStyles((theme: Theme) =>
+    createStyles({
+      root: {
+        textTransform: "none",
+        color: theme.palette.primary.main,
+        fontWeight: theme.typography.fontWeightBold,
+        fontSize: "16px",
+        opacity: 1,
+        marginRight: theme.spacing(1),
+        "&:focus": {
+          opacity: 1,
+          color: theme.palette.warning.dark,
+        },
+        "&:active": {
+          opacity: 1,
+          color: theme.palette.warning.dark,
+        },
+      },
+    })
+  )((props: StyledTabProps) => <Tab disableRipple {...props} />);
+
+  useEffect(() => {
+    function fetchBlogs() {
+      setJsonMdData(blogs);
+    }
+    fetchBlogs();
+  }, []);
+
+  var filteredAuthorData = (jsonMdData || []).filter(
+    (tabs: TabProps) => tabs.author === queryAuthorName
+  );
+
+
+  const handleChange = (_event: React.ChangeEvent<{}>, newValue: string) => {
+    setValue(newValue);
+  };
+
+  // functions to get the blog count for each individual tags
+  var totalBlogCount = (jsonMdData || []).filter(
```

Comment:
Same here, floating var in the namespace.

Reply:
changed it to **const**